otter/auth.py

```diff
@@ -284,6 +284,42 @@ def __hash__(self):
                      self._url, self._admin_url))
 
 
+@implementer(IAuthenticator)
+class SingleTenantAuthenticator(object):
+    """
+    An authentication handler that authenticates using a single tenant id.
+    """
+    def __init__(self, _identity_user, _identity_password, url):
```

Comment:
> If the user screws this up, wouldn't they just get authentication errors anyway?

Not really. Say the credentials are for tenant "t1" and someone calls `authenticate_tenant` on "t2", it will succeed and return token for "t1" which is unacceptable. 

> (Also, more importantly, if someone has a custom identity endpoint that lets certain creds authenticate for any tenant, this change would prevent them from using that for ostensibly little to no benefit.)

True. However, this code is authenticating to a keystone service which I am not sure allows that behavior. Even otherwise, I think another class can be implemented for that behavior.